tests/utils/node/node.go

```diff
@@ -0,0 +1,275 @@
+	}
+
+	if n.tomlConfig.Init.Genesis == "" {
+		n.tomlConfig.Init.Genesis = utils.GetGssmrGenesisRawPathTest(t)
+	}
+
+	if n.tomlConfig.Account.Key == "" {
+		keyList := []string{"alice", "bob", "charlie", "dave", "eve", "ferdie", "george", "heather", "ian"}
+		if *n.index < len(keyList) {
+			n.tomlConfig.Account.Key = keyList[*n.index]
+		} else {
+			n.tomlConfig.Account.Key = "default-key"
+		}
+	}
+
+	if n.tomlConfig.Network.Port == 0 {
+		const basePort uint16 = 7000
+		n.tomlConfig.Network.Port = basePort + uint16(*n.index)
+	}
+
+	if n.tomlConfig.RPC.Enabled && n.tomlConfig.RPC.Port == 0 {
+		const basePort uint32 = 8540
+		n.tomlConfig.RPC.Port = basePort + uint32(*n.index)
+	}
+
+	if n.tomlConfig.RPC.WS && n.tomlConfig.RPC.WSPort == 0 {
+		const basePort uint32 = 8546
+		n.tomlConfig.RPC.WSPort = basePort + uint32(*n.index)
+	}
+
+	userSetWriter := n.writer != nil && n.writer != io.Discard
+	if !userSetWriter {
+		n.logsBuffer = bytes.NewBuffer(nil)
+	}
+
+	if n.writer == nil {
+		n.writer = io.Discard
+	}
+
+	if n.binPath == "" {
+		n.binPath = pathfinder.GetGossamer(t)
+	}
+}
+
+// Init initialises the Gossamer node.
+func (n *Node) Init(ctx context.Context) (err error) {
+	cmdInit := exec.CommandContext(ctx, n.binPath, "init", //nolint:gosec
+		"--config", n.configPath,
+	)
+
+	if n.logsBuffer != nil {
+		n.writer = io.MultiWriter(n.writer, n.logsBuffer)
+	}
+
+	cmdInit.Stdout = n.writer
+	cmdInit.Stderr = n.writer
+
+	err = cmdInit.Start()
+	if err != nil {
+		return fmt.Errorf("cannot start command: %w", err)
+	}
+
+	err = cmdInit.Wait()
+	return n.wrapRuntimeError(ctx, cmdInit, err)
+}
+
+// Start starts a Gossamer node using the node configuration of
+// the receiving struct. It returns a start error if the node cannot
+// be started, and runs the node until the context gets canceled.
+// When the node crashes or is stopped, an error (nil or not) is sent
+// in the waitErrCh.
+func (n *Node) Start(ctx context.Context, waitErrCh chan<- error) (startErr error) {
+	cmd := exec.CommandContext(ctx, n.binPath, //nolint:gosec
+		"--config", n.configPath,
+		"--no-telemetry")
+
+	cmd.Stdout = n.writer
+	cmd.Stderr = cmd.Stdout // we assume no race between stdout and stderr
+
+	err := cmd.Start()
+	if err != nil {
+		return fmt.Errorf("cannot start %s: %w", cmd, err)
+	}
+
+	go func(cmd *exec.Cmd, node *Node, waitErr chan<- error) {
+		err = cmd.Wait()
+		waitErr <- node.wrapRuntimeError(ctx, cmd, err)
+	}(cmd, n, waitErrCh)
+
+	return nil
+}
+
+// StartAndWait starts a Gossamer node using the node configuration of
+// the receiving struct. It returns a start error if the node cannot
+// be started, and runs the node until the context gets canceled.
+// When the node crashes or is stopped, an error (nil or not) is sent
+// in the waitErrCh.
+// It waits for the node to respond to an RPC health call before returning.
+func (n *Node) StartAndWait(ctx context.Context, waitErrCh chan<- error) (startErr error) {
```

Comment:
Usually the function should return a `chan error`.  As a best practice, producers should return channels.

Reply:
I ended up making the changes, returning error channels instead of injecting them. It made things a bit more complicated, I had to add a 'errors fan in' object to handle stuff in `tests/utils/runtime/errors.go` but it did make responsibilities clearer, for example a consumer CANNOT close a channel (shame on my previous self!).